### PR TITLE
chore: Fix linter findings for `revive:enforce-slice-style` in `plugins/outputs`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Summary

Address findings for [revive:enforce-slice-style](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#enforce-slice-style) in `plugins/outputs`.

As we agreed here https://github.com/influxdata/telegraf/issues/15812, we will fix the errors found by the following configuration:
```
      - name: enforce-slice-style
        arguments: ["make"]
```

It is only part of the bigger job.
After all findings of this type in whole project are handled, we can enable `revive:enforce-slice-style` rule in `golangci-lint`.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/outputs/graylog/graylog.go [hunk 1]
```diff
@@ -377,7 +377,7 @@ func (g *Graylog) connectRetry(tlsCfg *tls.Config) {
 
 	g.wg.Add(1)
 
-	unconnected := append([]string{}, g.Servers...)
+	unconnected := g.Servers
```
Comment (DStrand1) [inline plugins/outputs/graylog/graylog.go:380]: Could we just use `g.Servers` directly or does it need to be a new variable?

**Review by DStrand1 — COMMENTED** [review]
Overall looks great! Just one small comment from my end

File: plugins/outputs/graylog/graylog.go (hunk 1 above)
Comment (srebhan) [reply plugins/outputs/graylog/graylog.go:380]: It must be `append` and with the current code, modifying `unconnected` will modify `g.Server` which **must** be avoided here!

Comment (zak-pawel) [reply plugins/outputs/graylog/graylog.go]: Fixed.

**Review by DStrand1 — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Nice. Thank you @zak-pawel!

**Pull request merged**